src/test/java/core/basesyntax/serviceimpltest/TransactionParseServiceImplTest.java

```diff
@@ -0,0 +1,176 @@
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class TransactionParseServiceImplTest {
+    private static final int COLUMNS_LENGTH = 3;
+    private static final String SEPARATOR = ",";
+    private static final String FIRST_COLUMN = "type";
+    private static final String SECOND_COLUMN = "fruits";
+    private static final String THIRD_COLUMN = "quantity";
+    private static TransactionParseService transactionParseService;
+    private FruitShopService fruitShopService;
+    private FruitTransaction fruitTransactionBalance;
+    private FruitTransaction fruitTransactionSupply;
+    private FruitTransaction fruitTransactionPurchase;
+    private FruitTransaction fruitTransactionReturn;
+
+    @BeforeEach
+    void setUp() {
+        transactionParseService = new TransactionParseServiceImpl();
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE,
+                new BalanceOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN,
+                new ReturnOperationHandler());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+        fruitShopService = new FruitsShopServiceImpl(operationStrategy);
+        fruitTransactionBalance = new FruitTransaction(
+                FruitTransaction.Operation.BALANCE, "apple", 20);
+        fruitTransactionPurchase = new FruitTransaction(
+                FruitTransaction.Operation.PURCHASE, "apple", 20);
+        fruitTransactionSupply = new FruitTransaction(
+                FruitTransaction.Operation.SUPPLY, "apple", 20);
+        fruitTransactionReturn = new FruitTransaction(
+                FruitTransaction.Operation.RETURN, "apple", 20);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.FRUIT_MAPS.clear();
+    }
+
+    @Test
+    public void parseDataWithInvalidRow_Ok() {
+        String invalidData = "type,fruits,quantity\n"
+                + "b,apple,20\n"
+                + "###,Orange,invalid";
+        assertThrows(RuntimeException.class, () -> transactionParseService.parse(invalidData));
+    }
+
+    @Test
+    void type_nullData_notOk() {
+        assertThrows(RuntimeException.class, () -> transactionParseService.parse(null));
+    }
+
+    @Test
+    void first_column_Ok() {
+        assertTrue(true, FIRST_COLUMN);
```

Comment:
this assertion is always okay and tests anything. 
```
assertTrue(boolean condition, String message)
Asserts that the supplied condition is true.
```

the same for `assertFalse(false)` and all test below.